tools/dkcm/mainservice/mainservice.go

```diff
@@ -0,0 +1,146 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mainservice
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"knative.dev/test-infra/tools/dkcm/clerk"
+
+	"knative.dev/test-infra/pkg/mysql"
+)
+
+var (
+	dbConfig             *mysql.DBConfig
+	db                   *clerk.DB
+	DefaultClusterParams = clerk.ClusterParams{Zone: DefaultZone, Nodes: 4, NodeType: DefaultNodeType}
+)
+
+// handle cleaning cluster request after usage
+func handleCleanCluster(w http.ResponseWriter, r *http.Request) {
+
+}
+
+// check the pool capacity
+func checkPoolCap(cp *clerk.ClusterParams) {
+	numAvail := db.CheckNumAvail(cp)
+	if numAvail < DefaultOverProvision {
+		// create cluster if not meeting overprovisioning criteria
+		CreateCluster()
+	}
+}
+
+// assign clusters if available upon request
+func CreateCluster() {
+
+}
+
+// assign clusters if available upon request
+func AssignCluster(token string, w http.ResponseWriter) {
+	r, err := db.GetRequest(token)
+	if err != nil {
+		http.Error(w, "there is an error getting the request with the token. Please try again.", http.StatusNotAcceptable)
+		return
+	}
+	available, clusterID := db.CheckAvail(r.ClusterParams)
+	if available == true {
+		response, err := db.GetCluster(clusterID)
+		if err != nil {
+			http.Error(w, "there is an error getting available clusters. Please try again.", http.StatusInternalServerError)
+			return
+		}
+		db.UpdateRequest(r.RequestID, clerk.UpdateNumField("ClusterID", clusterID))
+		checkPoolCap(&DefaultClusterParams)
+		w.Write([]byte("Your cluster is ready! Here is the info:\n"))
+		w.Write([]byte(ReturnCluster(response)))
+	}
+	w.Write([]byte("Your cluster isn't ready yet! Please check back later.\n"))
+
+}
+
+// return cluster info
+func ReturnCluster(response *clerk.Response) string {
+	return fmt.Sprintf("ClusterName: %s, ProjectID: %s, Zone: %s", response.ClusterName, response.ProjectID, response.Zone)
+}
+
+// main go
+func LoadDB() {
+
+}
+
+func GetProject() {
+
+}
+
+func ReleaseProject() {
+
+}
+
+// handle new cluster request
+func handleNewCluster(w http.ResponseWriter, req *http.Request) {
+	prowJobID := req.PostFormValue("prowjobid")
+	nodesCount, err := strconv.Atoi(req.PostFormValue("nodes"))
+	if err != nil {
+		nodesCount = DefaultNodesCount
+	}
+	nodesType := req.PostFormValue("nodeType")
+	zone := req.PostFormValue("zone")
+	cp := clerk.NewClusterParams(clerk.AddZone(zone), clerk.AddNodes(int64(nodesCount)), clerk.AddNodeType(nodesType))
+	r := clerk.NewRequest(clerk.AddProwJobID(prowJobID), clerk.AddrequestTime(time.Now()))
+	r.ClusterParams = cp
+	accessToken, err := db.InsertRequest(r)
+	if err != nil {
+		http.Error(w, "there is an error creating new request. Please try again.", http.StatusInternalServerError)
+		return
+	}
+	w.Write([]byte(fmt.Sprintf("accessToken: %s", accessToken)))
```

Comment:
The response could be only the accessToken:
```suggestion
	w.Write([]byte(accessToken))
```